### ISSUE TITLE
Realm file upload, can't able to process the data inside it, ( no realm export and import ) in React Native

### ISSUE THREAD

**Issue by indranilbhuin** [realm/realm-js#6332]
### Problem
There is no official way to export or import the realm data. I am able to download the .realm file but can't able to retrieve the data from that.

this is the export function

```import
import RNFS from 'react-native-fs';
import {getRealm} from './realmService';

export const exportRealmData = async () => {
  try {
    const realm: Realm = await getRealm();
    const exportFileName = 'qqqq.realm';
    const exportFilePath = `${RNFS.DocumentDirectoryPath}/${exportFileName}`;

    if (await RNFS.exists(exportFilePath)) {
      await RNFS.unlink(exportFilePath);
    }

    const config = {
      path: exportFilePath,
      schema: realm.schema,
      schemaVersion: realm.schemaVersion + 1,
      encryptionKey: realm.encryptionKey,
    };

    realm.writeCopyTo(config);

    const downloadsFolder = RNFS.DownloadDirectoryPath;
    const destinationFilePath = `${downloadsFolder}/${exportFileName}`;

    await RNFS.moveFile(exportFilePath, destinationFilePath);

    console.log('Realm data exported to', destinationFilePath);
    return destinationFilePath;
  } catch (error) {
    console.error('Error exporting Realm data', error);
  }
};
```

I have tried to upload and retrieve the data, but i am unable to do that

      function normalizePath(path: string | undefined) {
        try {
          if (path === undefined) {
            throw console.error('undefined');
          }
          if (Platform.OS === 'ios' || Platform.OS === 'android') {
            const filePrefix = 'file:';
            if (path.startsWith(filePrefix)) {
              path = path.substring(filePrefix.length);
            }
            path = decodeURIComponent(path);
            return path;
          }
        } catch (e) {
          console.error({msg: 'Failed to normalize path', data: e});
        }
      }
    
      const handleFilePick = async () => {
        try {
          const result = await DocumentPicker.pick({
            type: [DocumentPicker.types.allFiles],
          });
          console.log(result);
          const {0: file1} = result;
    
          const path = normalizePath(file1?.uri);
          console.log(path);
          setRealmFile(path);
        } catch (err) {
          console.error(err);
        }
      };
    
      console.log(realmFile);
    
      const handleDataSubmit = async () => {
        try {
          if (!realmFile) {
            console.error('Please upload a .realm file');
            return;
          }
    
          const realm = await getRealm();
          const existingRealm = await Realm.open({path: realmFile.uri});
    
          console.log('This is the existing realm:', existingRealm);
    
          realm.write(() => {
            [User, Category, Expense, Currency, Debtor, Debt].forEach(schema => {
              const objects = existingRealm.objects(schema.schema.name);
              console.log(`Data for ${schema.schema.name}:`, objects.toJSON());
    
              objects.forEach(object => {
                realm.create(schema.schema.name, object, Realm.UpdateMode.Modified);
              });
            });
          });
    
          await AsyncStorageService.setItem('isOnboarded', JSON.stringify(true));
          dispatch(setIsOnboarded(true));
    
          console.log('Data updated successfully!');
        } catch (error) {
          console.error('Error updating data:', error);
        }
      };This is the log

```
LOG  content://com.android.providers.downloads.documents/document/msf:1000000035
 LOG  [{"fileCopyUri": null, "name": "qqqq.realm", "size": 4848, "type": "application/octet-stream", "uri": "content://com.android.providers.downloads.documents/document/msf%3A1000000035"}]
 LOG  content://com.android.providers.downloads.documents/document/msf:1000000035
 LOG  This is the existing realm: {"beforeNotifyListeners": {"eventType": "beforenotify", "listeners": Set {}, "realm": [Circular]}, "changeListeners": {"eventType": "change", "listeners": Set {}, "realm": [Circular]}, "schemaExtras": {}, "schemaListeners": {"eventType": "schema", "listeners": Set {}, "realm": [Circular]}, "syncSession": null}
```

### Solution
a simple realm.export and realm.import function will be easy to interact with the data.

### How important is this improvement for you?
Would be a major improvement

### Feature would mainly be used with
Local Database only

**Comment by kneth** [comment]
@indranilbhuin Thank you for the suggestion.

In version 11.0.0 we changed our `toJSON()` methods to work with libraries like [flatted](https://www.npmjs.com/package/flatted) as Realm database can potentially have cycles. Our reasoning was that `toJSON()` could we used by our users to export objects, and [flatted](https://www.npmjs.com/package/flatted) could be a way to import data by transforming a serialized format to plain JavaScript objects which can be used with `Realm.create()`.

**Comment by indranilbhuin** [comment]
Hey @kneth

Thank you for your prompt response and clarification regarding the changes made in version 11.0.0, especially the shift in the toJSON() methods to accommodate libraries like [flatted](https://www.npmjs.com/package/flatted). I appreciate the reasoning behind this adjustment, as it certainly addresses potential challenges with cyclic dependencies in Realm databases.

I've made an effort to implement the suggested approach using flatted for serialization and deserialization. However, I encountered difficulties in successfully deserializing and retrieving data from the .realm file. After some troubleshooting, I decided to revise the export logic by opting for a workaround: exporting data as a .json file and then utilizing `realm.create()` to populate the local database. While this solution works, I believe that having an inbuilt Realm function for export and import would greatly enhance the user experience and streamline data management processes.

this is the import logic
```
const importRealmData = async () => {
  try {
      const result = await DocumentPicker.pick({
        type: [DocumentPicker.types.allFiles],
      });

      const {0: res} = result;
      const path = normalizePath(res.uri);
      if (path === undefined) {
        throw console.error('undefined');
      }

      console.log(result);
      console.log('this is the path', res.uri, path);

      const fileContent = await RNFS.readFile(path, 'utf8');

      const jsonData = JSON.parse(fileContent);

      const {key, data} = jsonData;
      setAllData(data);

      console.log('Key:', key);
      console.log('Data:', data);
    } catch (error) {
      console.error('Error picking or reading file:', error);
    }
  };
```
An inbuilt export-import functionality within Realm would not only simplify data transfer but also align with the seamless nature of the Realm database. I believe such a feature would be beneficial for users who need to migrate data or perform backups efficiently.

Thank you once again for your time and assistance.
